47.md

```diff
@@ -373,6 +373,34 @@ Response:
 }
 ```
 
+### `get_budget`
+
+Budgets can be used to limit the amount of funds that can be spent by a connection in a given time period.
+Budgets are optional and can be set by the user if their wallet supports it. Support for this command indicates
+that the wallet supports user-configured budgets. The budget will be reset at the `renews_at` timestamp.
+
+Request:
+```jsonc
+{
+    "method": "get_budget",
+    "params": {
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "get_budget",
+    "result": {
+        "remaining_budget_msats": 10000,
+        "total_budget_msats": 100000,
```

Comment:
nowhere else in the spec we suffix with `msats` - we use msats for all values as per the comments on the other command request and responses

Reply:
Yeah, that's true. Transparently, I'm being explicit here because I'm setting up plans to get this to work reasonably with https://github.com/nostr-protocol/nips/pull/1353. My plan for cross-currency budgets is something like:

```json
{
    "result_type": "get_budget",
    "result": {
        "remaining_budget_msats": 10000,
        "total_budget_msats": 100000,
        "renews_at": 1693876973, // timestamp in seconds since epoch, optional. If not provided, the budget does not renew.
        "renewal_period": "monthly", // daily|weekly|monthly|yearly|never
        "currency": {
            "name": "US Dollar",
            "code": "USD",
            "symbol": "$",
            "decimals": 2,
            "total_budget": 1000,
            "remaining_budget": 100,
        }
    }
}
```

That way, we can be explicit about the top-level budget being in msats while the user-facing currency is inside the `currency` struct.

Reply:
The currency of NWC is bitcoin and the unit is msats, this is the case for all requests and responses. To me it makes more sense for the optional currency extensions to be explicit on their currency rather than introducing _msats on the core protocol which then makes the budget method inconsistent with all the existing methods.

Reply:
Fair enough. Will remove for now and consider other options for multi-currency stuff.